contents/blog/product-engineer-vs-software-engineer.md

```diff
@@ -0,0 +1,66 @@
+
+Software engineers focus on writing code, testing it, and maintaining it. They  look for ways to optimize code, improve scalability, and solve bugs. They are deeply focused on specific areas of technology, whether that is databases, data pipelines, backend APIs, or client-side app frameworks. This means researching and reading documentation, updates, and code.
+
+## What do their teams and companies look like?
+
+Product engineers are often a part of smaller teams, where they must do more themselves. Startups, and earlier-stage companies that require people to do more, are often looking for product engineers (even if they don’t say it). They need people who can own large decisions and ship quickly. Product engineers fit this need.
+
+Software engineers are part of larger teams, usually spread out across multiple teams. They have a team to support them, so they can specialize. Their team might include designers, product managers, and marketers. This allows them to think less about if what they are building is right, and more about the implementation.
+
+At the end of the day, it’s up to the company and team to decide what roles they need. There are always debates about what the right structure and right roles are. Product engineers and software engineers both have their place in the tech world. It all comes down to the goals of the team and the needs of the customers.
+
+## What’s right for you?
+
+As a person, if owning a product and its roadmap as well as responsibility for its success or failure interests you, then product engineering might be for you. If you want to focus on solving difficult technical problems and optimizing existing technologies and processes, then software engineering might be a better choice.
+
+There isn’t a right answer, and roles change as companies change. It is possible a product engineer to focus on a specific area and becomes a software engineer as a company grows. It’s also possible for software engineers to join new teams or companies and take on a product engineering role.
+
+As a company, having software engineers appeals to companies with a large-scale or complex products. Software engineers can focus deeply on specific areas of technology and are great for optimizing systems.
+
+Startups or companies that want to be agile and move fast, should look for product engineers. Product engineers have a holistic view of the product and the ability to move fast when the need arises.
+
+Picking what is right is up to the company and its culture. If focusing on building great products is a priority, product engineers might be a better choice. If scaling and building expertise throughout the company is a focus, then software engineers are likely better. Companies with both types of roles have succeeded and more will continue to succeed.
```

Comment:
I think we need to be a bit more opinionated here, particularly around the skills needed to be a successful product engineer. The biggest point, I suspect, is you need to comfortable talking to users so I think we should say this outright.

Reply:
It is more opinionated now.